E0527 needs to be updated to new format

From: src/test/compile-fail/E0527.rs

E0527 needs a span_label, updating it from:

```
error[E0527]: pattern requires 2 elements but array has 4
  --> src/test/compile-fail/E0527.rs:16:10
   |
16 |         &[a, b] => { //~ ERROR E0527
   |          ^^^^^^
```

To:

```
error[E0527]: pattern requires 2 elements but array has 4
  --> src/test/compile-fail/E0527.rs:16:10
   |
16 |         &[a, b] => { //~ ERROR E0527
   |          ^^^^^^ expected 4 elements
```

--- a/src/librustc_typeck/check/_match.rs
+++ b/src/librustc_typeck/check/_match.rs
@@ -233,9 +233,12 @@ impl<'a, 'gcx, 'tcx> FnCtxt<'a, 'gcx, 'tcx> {
                         let min_len = before.len() + after.len();
                         if slice.is_none() {
                             if min_len != size {
-                                span_err!(tcx.sess, pat.span, E0527,
-                                          "pattern requires {} elements but array has {}",
-                                          min_len, size);
+                                struct_span_err!(
+                                    tcx.sess, pat.span, E0527,
+                                    "pattern requires {} elements but array has {}",
+                                    min_len, size)
+                                    .span_label(pat.span, &format!("expected {} elements",size))
+                                    .emit();
                             }
                             (inner_ty, tcx.types.err)
                         } else if let Some(rest) = size.checked_sub(min_len) {

--- a/src/test/compile-fail/E0527.rs
+++ b/src/test/compile-fail/E0527.rs
@@ -13,7 +13,9 @@
 fn main() {
     let r = &[1, 2, 3, 4];
     match r {
-        &[a, b] => { //~ ERROR E0527
+        &[a, b] => {
+            //~^ ERROR E0527
+            //~| NOTE expected 4 elements
             println!("a={}, b={}", a, b);
         }
     }

